lib/cisco_node_utils/vlan.rb

```diff
@@ -53,19 +61,33 @@ def destroy
       config_set('vlan', 'destroy', @vlan_id)
     end
 
-    def cli_error_check(result)
+    def cli_error_check(result, ignore_message=nil)
       # The NXOS vlan cli does not raise an exception in some conditions and
       # instead just displays a STDOUT error message; thus NXAPI does not detect
       # the failure and we must catch it by inspecting the "body" hash entry
       # returned by NXAPI. This vlan cli behavior is unlikely to change.
+      # Check for messages that can be safely ignored.
+      unless ignore_message.nil?
+        # Add logic to check results against ignore_message here
+        # Return if results == ignore_message
+        return if
+             result[2].is_a?(Hash) &&
+             ignore_message[0] == result[2]['body'].to_s ||
+             ignore_message[1] == result[2]['body'].to_s
+        return if
+             result[2].is_a?(String) &&
+             ignore_message[0] == result[2]['body'].to_s ||
+             ignore_message[1] == result[2]['body'].to_s
+      end
+
       fail result[2]['body'] if
         result[2].is_a?(Hash) &&
-        /(ERROR:|Warning:)/.match(result[2]['body'].to_s)
+        /(ERROR:|Warning:|VLAN:)/.match(result[2]['body'].to_s)
```

Comment:
A primary vlan can have only one isolated vlan associated to it. When a user tries to associate another isolated vlan, an error message should be generated. On N7k there is no ERROR: at the start of  the message but VLAN:
In my opinion we should change the N7k behavior but for time being we need this check for VLAN: